tiledb/sm/serialization/group.cc

```diff
@@ -336,15 +336,15 @@ Status group_create_details_to_capnp(
     return LOG_STATUS(
         Status_SerializationError("Error serializing group; group is null."));
   }
-  if (!legacy) {
-    // TileDB-Server requests do not support custom storage locations.
-    return Status::Ok();
-  }
 
   const auto& group_uri = group->group_uri();
   if (group_uri.is_tiledb()) {
     URI::RESTURIComponents rest_uri;
     RETURN_NOT_OK(group->group_uri().get_rest_components(legacy, &rest_uri));
+    if (!legacy && !rest_uri.asset_storage.empty()) {
+      return LOG_STATUS(Status_SerializationError(
```

Comment:
I don't like it too much that we throw a "serialization error", it's user input error on creating a group (or array). Either we should throw another type of move the check up into the rest client and throw from there.